playbooks/aws/openshift-cluster/tasks/launch_instances.yml

```diff
@@ -34,6 +35,32 @@
     ec2_assign_public_ip: "{{ lookup('env', 'ec2_assign_public_ip')
                     | default(deployment_vars[deployment_type].assign_public_ip, true) }}"
   when: ec2_assign_public_ip is not defined
+
+- set_fact:
+    ec2_instance_type: "{{ ec2_master_instance_type }}"
+  when: host_type == "master"
+- set_fact:
+    ec2_security_groups: "{{ ec2_master_security_groups }}"
+  when: host_type == "master"
+
+- set_fact:
+    ec2_instance_type: "{{ ec2_infra_instance_type }}"
+  when: sub_host_type == "infra"
+- set_fact:
+    ec2_security_groups: "{{ ec2_infra_security_groups }}"
+  when: sub_host_type == "infra"
+
+- set_fact:
+    ec2_instance_type: "{{ ec2_node_instance_type }}"
+  when: sub_host_type == "node"
```

Comment:
@wshearn correct, that would be the proper way to do it in ansible, I was just providing a generic pseudo code outline for how I thought the logic should work.

closer to ansible proper it would be something like this:
```
set_fact:
  <master_facts>
when: host_type == "master"

set_fact:
  <etcd_facts>
when: host_type == "etcd"

set_fact:
  <infra node facts>
when: host_type == "node" and sub_host_type == "infra"

set_fact:
  <compute host facts>
when: host_type == "node" and sub_host_type == "compute"
```